packages/dataviews/src/components/dataform-context/index.tsx

```diff
@@ -0,0 +1,53 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import type { FormField, NormalizedField } from '../../types';
+
+type DataFormContextType< Item > = {
+	getFieldDefinition: (
+		field: FormField
+	) => NormalizedField< Item > | undefined;
+};
+
+const DataFormContext = createContext< DataFormContextType< any > >( {
+	getFieldDefinition: () => undefined,
+} );
+
+export function DataFormProvider< Item >( {
+	fields,
+	children,
+}: React.PropsWithChildren< { fields: NormalizedField< Item >[] } > ) {
+	const getFieldDefinition = useCallback(
```

Comment:
Can't we move this logic to `normalizeFields` function? In this way, we can get rid of this context